lib/elixir_sense/core/normalized/code.ex

```diff
@@ -93,4 +90,55 @@ defmodule ElixirSense.Core.Normalized.Code do
   defp extract_docs(%{"en" => docs_en}), do: docs_en
   defp extract_docs(:hidden), do: false
   defp extract_docs(:none), do: nil
+
+  defp get_fun_docs(module, docs) do
```

Comment:
@msaraiva are you OK with introducing custom changes to this module? Before that all code in `Normalized` used to be a thin portability wrapper over stdlib stuff

Reply:
In case it helps, iex [does something similar](https://github.com/elixir-lang/elixir/blob/27bd9ffcc607b74ce56b547cb6ba92c9012c317c/lib/iex/lib/iex/introspection.ex#L458-L462). I presume that ex_doc also does this (but didn't test it).

Reply:
> code in `Normalized` used to be a thin portability wrapper over stdlib stuff

Indeed. That's was the intention. At some point, especially between v1.4 and v1.7, it was really hard to keep track of conditionals related to different versions of Elixir and then at some point, I decided to search for them in the code and move everything to `Normalized`. My dream was to someday the language would stabilize and I'd be able to remove the whole thing :)

I believe that day has come. I took a quick look at all the functions there and I think it's time to get rid of them and start using only the newer built-in counterparts. As far as I could see, we can do that if we drop support for Elixir < v1.7.

Regarding @sasa1977's changes, I think we can leave there for now and then, when we finally get rid of  `Normalized.Code`, we move them to someplace else.

Just to make it clear, I think we can still have `Normalized` for other things that might appear in the near future, however, keeping them for more than 2 minor updates makes it harder to maintain the rest of the code. Since Elixir is already on v1.10, I guess we are allowed to let it go ;)

WDYT?

Reply:
>  keeping them for more than 2 minor updates makes it harder to maintain the rest of the code

I'm not completely sure about the circumstances of this project, but in general my position on OSS, especially if it's maintained in people's own free time is that it's a "best effort". What this means in the context of Elixir versions is that people who want to use this lib on older Elixir versions can always lock to some past sha (having a tag provided would be nice). The new versions of the lib should be free to upgrade their lang version requirements. I think this is particularly fair in Elixir land, because upgrades to both Elixir & Erlang have in my experience always been quite painless. But even if they weren't, I'd have the same position. It's hard enough to maintain a lib for a single version of Elixir, so there have to be some strong reasons to support multiple mutually incompatible versions.

In other words, clients who want new features need to move to new Elixir (& Erlang if required). The clients who want old Elixir/Erlang compatibility won't get new features. I believe that for a software which is free of charge, this is a reasonable trade-off :-)

Reply:
> In other words, clients who want new features need to move to new Elixir (& Erlang if required). The clients who want old Elixir/Erlang compatibility won't get new features. I believe that for a software which is free of charge, this is a reasonable trade-off :-)
It's reasonable but I'd prefer to avoid python 3 mess.